### ISSUE TITLE
Ability to use custom docker images for CodeBuild

### ISSUE THREAD

**Issue by eriklz** [awslabs/aws-deployment-framework#196]
We have a requirement to run a couple of different custom docker images in some of our pipelines. This includes images fetched from ECR and images from DockerHub.

From what I could see in ADF 2.1.2 it does not support any images besides the predefined CodeBuild images currently. SO it would be great to be able to specify ECR images as well as DockerHub images, for at least build stage with CodeBuild.

**Comment by bundyfx** [comment]
Thanks @eriklz 

Tested this out today in 3.0 beta and is working as intended and will make its way into the official 3.0 release shortly.

**Comment by amitsehgal** [comment]
@bundyfx looks like standard image provided by aws is also erroring out. Does above fix takes care of that ? tried using AMAZON_LINUX_2_2 https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-codebuild.LinuxBuildImage.html
```Traceback (most recent call last):
  File "adf-build/cdk/generate_pipeline_stacks.py", line 37, in <module>
    main()
  File "adf-build/cdk/generate_pipeline_stacks.py", line 33, in main
    PipelineStack(app, stack_input)
  File "/codebuild/output/src189630871/src/adf-build/jsii/_runtime.py", line 66, in __call__
    inst = super().__call__(*args, **kwargs)
  File "/codebuild/output/src189630871/src/adf-build/cdk/cdk_stacks/main.py", line 69, in __init__
    {} # Empty target since this is a build only stage
  File "/codebuild/output/src189630871/src/adf-build/jsii/_runtime.py", line 66, in __call__
    inst = super().__call__(*args, **kwargs)
  File "/codebuild/output/src189630871/src/adf-build/cdk/cdk_constructs/adf_codebuild.py", line 83, in __init__
    build_image=getattr(_codebuild.LinuxBuildImage, map_params['default_providers']['build'].get('properties', {}).get('image', "UBUNTU_14_04_PYTHON_3_7_1")),
AttributeError: type object 'LinuxBuildImage' has no attribute 'AMAZON_LINUX_2_2'
Subprocess exited with error 1```

**Comment by bundyfx** [comment]
Hi @amitsehgal 

Yes that is also resolved in 3.0 - thats because of the cdk version being used in 2.1.2 doesnt know about that image. we're bumped to 1.2.0 in 3.0.

**Comment by sbkok** [comment]
Thank you for your patience. I am happy to inform you that this feature has been [released as part of v3.2.0](https://github.com/awslabs/aws-deployment-framework/releases/tag/v3.2.0) just now.
Please open a new issue if you are experiencing any issues related to this feature.
